mangum/adapter.py

```diff
@@ -91,89 +91,108 @@ def __call__(self, event: dict, context: "LambdaContext") -> dict:
-                query_string = event.get("rawQueryString", "").encode()
+        return response
 
-                if event.get("cookies"):
-                    headers["cookie"] = "; ".join(event.get("cookies", []))
+    def event_to_request_body(self, event: dict) -> bytes:
+        """Extracts request body from a Lambda event."""
+        is_binary = event.get("isBase64Encoded", False)
+        initial_body = event.get("body") or b""
+        if is_binary:
+            initial_body = base64.b64decode(initial_body)
+        elif not isinstance(initial_body, bytes):
+            initial_body = initial_body.encode()
 
-            # API Gateway v1 / ELB
-            else:
-                if "elb" in request_context:
-                    # NOTE: trust only the most right side value
-                    source_ip = headers.get("x-forwarded-for", "").split(", ")[-1]
-                else:
-                    source_ip = request_context.get("identity", {}).get("sourceIp")
-
-                path = event["path"]
-                http_method = event["httpMethod"]
-
-                if event.get("multiValueQueryStringParameters"):
-                    query_string = urllib.parse.urlencode(
-                        event.get("multiValueQueryStringParameters", {}), doseq=True
-                    ).encode()
-                elif event.get("queryStringParameters"):
-                    query_string = urllib.parse.urlencode(
-                        event.get("queryStringParameters", {})
-                    ).encode()
-                else:
-                    query_string = b""
-
-            server_name = headers.get("host", "mangum")
-            if ":" not in server_name:
-                server_port = headers.get("x-forwarded-port", 80)
-            else:
-                server_name, server_port = server_name.split(":")  # pragma: no cover
-            server = (server_name, int(server_port))
-            client = (source_ip, 0)
-
-            if not path:  # pragma: no cover
-                path = "/"
-            elif self.api_gateway_base_path:
-                if path.startswith(self.api_gateway_base_path):
-                    path = path[len(self.api_gateway_base_path) :]
-
-            scope = {
-                "type": "http",
-                "http_version": "1.1",
-                "method": http_method,
-                "headers": [[k.encode(), v.encode()] for k, v in headers.items()],
-                "path": urllib.parse.unquote(path),
-                "raw_path": None,
-                "root_path": "",
-                "scheme": headers.get("x-forwarded-proto", "https"),
-                "query_string": query_string,
-                "server": server,
-                "client": client,
-                "asgi": {"version": "3.0"},
-                "aws.event": event,
-                "aws.context": context,
+        return initial_body
+
+    def event_to_asgi_scope(self, event: dict) -> Scope:
```

Comment:
Name this `create_scope`.

Reply:
done. Since that's a private method - renamed to `_create_scope`